utils.groovy

```diff
@@ -282,4 +282,110 @@ def get_artifacts_to_build(pipecfg, stream, basearch) {
     }
     return artifacts.unique()
 }
+// Upload artifacts to clouds
+def upload_to_clouds(pipecfg, basearch, buildID, stream) {
+    // Get artifacts to upload
+    def meta_json = "builds/${buildID}/${basearch}/meta.json"
+    def meta = readJSON file: meta_json
+    def artifacts = meta['images'].keySet() as List
+    def clouds = pipecfg.clouds.keySet() as List
+    // Check if we have cloud config to upload these artifacts
+    def uploads = artifacts.intersect(clouds)
+    parallel uploads.inject([:]) { branch, cloud -> branch["Upload ${cloud}"] = {
+
+        def var_name = cloud.toUpperCase()
+        tryWithCredentials([file(variable: "${var_name}_BUILD_UPLOAD_CONFIG",
+                                 credentialsId: "${cloud}-build-upload-config")]) {
```

Comment:
unfortunately this isn't 100% effective.

For AWS we use `aws-build-upload-config` but for all other clouds we've been using `"${cloud}-image-upload-config"`.

For AWS we're uploading the build artifacts (hence `aws-build-upload-config` name) and also an AMI. We just happened to re-use the secret for the AMI creation. For others we're only creating the images, so they were named appropriately.

Reply:
Does it make sense to have 2 secrets to `aws` so? So we can match the names, or even change it to match the others?

Reply:
Yes, It could make sense to have a separate secret here, but I kind of think we should only do that if we plan to create AMIs in a different account that where we are doing the build upload. I guess maybe that's the case for China or GovCloud?